### ISSUE TITLE
(aws-backup): BackupResource with TagCondition is not intuitive

### ISSUE THREAD

**Issue by pagelypete** [aws/aws-cdk#18911]
### What is the problem?

When using `BackupResource` the behaviour especially with `fromResource()` and a `TagCondition` is not intuitive. It produces results that are the union of the two conditions rather than results matching both conditions.

### Reproduction Steps

For example, the following code:

```javascript
        myBackupPlan.addSelection('SomeVolumes', {
            resources: [
                new backup.BackupResource('arn:aws:ec2:*:*:volume/*', {
                  key: "aws:ResourceTag/aws-backup",
                  value: "1",
                  operation: backup.TagOperation.STRING_EQUALS
                })
            ],
        });
```

I would expect that to backup all EBS volumes that also have the tag `aws-backup` = `1` since the tag condition is actually included as part of the resource.

Instead, it actually produces the following in the backup selection:

```
    "BackupSelection": {
        "SelectionName": "SomeVolumes",
        "IamRoleArn": "xxx",
        "Resources": [
            "arn:aws:ec2:*:*:volume/*"
        ],
        "ListOfTags": [
            {
                "ConditionType": "STRINGEQUALS",
                "ConditionKey": "aws:ResourceTag/aws-backup",
                "ConditionValue": "1"
            }
        ]
    },
```

Which means it backs up all EBS volumes and all resources (EBS or not) with `aws-backup` = `1`. I would expect it to instead produce `Conditions` in the output and not `ListOfTags` at all.

If I wanted all volumes and all resources tagged `aws-backup` = `1` I would expect to use the following (which does work as expected):

```javascript
        myBackupPlan.addSelection('SomeVolumes', {
            resources: [
                new backup.BackupResource.fromArn('arn:aws:ec2:*:*:volume/*'),
                new backup.BackupResource.fromTag('aws-backup', '1')
            ],
        });
```

In addition, it would also be very nice to have the ability to specify more than one TagCondition when using a `BackupSelection` since it seems otherwise there is no way to access `Conditions` at all without resorting to a `Cfn` level construct.

### What did you expect to happen?

`Conditions` to be used to ensure the resources meets **ALL** conditions not any condition.

### What actually happened?

Resources are selected based on any matching condition.

### CDK CLI Version

2.10.0 (build e5b301f)

### Node.js Version

v16.14.0

### OS

Ubuntu

### Language

Typescript

### Language Version

JavaScript

**Comment by javydekoning** [comment]
I'm running into the same issue today. 

ListOfTags uses OR logic, so defined resources OR any resource that matches tag. This should default to conditions indeed.

**Comment by thiagosantosleite** [comment]
Looks like TagOperation is limited to STRINGEQUALS:
![image](https://user-images.githubusercontent.com/1839326/212069585-65f20546-8630-45d2-9046-65221569db5b.png)

In the console/API we have much more options:
![image](https://user-images.githubusercontent.com/1839326/212069697-d3375408-f768-4586-956d-a056f4de9be5.png)

I would like to add a selections based on tagkey. I tried this, but does not works:
```
databaseBackupPlan.addSelection('DatabaseBackupSelection', {
      backupSelectionName: 'backup-all-tenants',
      resources: [
        backup.BackupResource.fromArn('arn:aws:rds:*:*:cluster:*'),
        backup.BackupResource.fromTag('aws:TagKeys', 'backup'),
      ],
    });
```

would be nice if we have more TagOperation options available.

**Comment by bstascavage** [comment]
Agreed.  I am trying to exclude resources from a selection and I currently can't because of the limited use of `TagCondition`

**Comment by Spjak** [comment]
Facing the same issue here.

Specifically, it seems like there is no way to specify the `Conditions` property on the BackupSelection, it will always generate the `ListOfTags` based on the provided BackupResources.

In the CloudFormation documentation [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-backup-backupselection-backupselectionresourcetype.html) it states:
> Conditions differs from ListOfTags as follows:
> When you specify more than one condition, you only assign the resources that match ALL conditions (using AND logic).

> ListOfTags differs from Conditions as follows:
> When you specify more than one condition, you assign all resources that match AT LEAST ONE condition (using OR logic).

It does not seem to matter how I construct my BackupResource in CDK - it always generates ListOfTags.
```
const myResource = new BackupResource('arn:*:fsx:*', {
    key: 'aws:ResourceTag/automaticBackup',
    value: 'true',
    operation: TagOperation.STRING_EQUALS
})

const myOtherResource = BackupResource.fromTag('aws:ResourceTag/automaticBackup', 'true', TagOperation.STRING_EQUALS)

backupPlan.addSelection('FSxBackup', {
    resources: [
        myResource,
        myOtherResource
    ]
})
```
Results in 
```
"BackupSelection": {
   "IamRoleArn": {
    "Fn::GetAtt": [
     "BackupPlanFSxBackupRole16A4E2D7",
     "Arn"
    ]
   },
   "ListOfTags": [
    {
     "ConditionKey": "aws:ResourceTag/automaticBackup",
     "ConditionType": "STRINGEQUALS",
     "ConditionValue": "true"
    },
    {
     "ConditionKey": "aws:ResourceTag/automaticBackup",
     "ConditionType": "STRINGEQUALS",
     "ConditionValue": "true"
    }
   ],
   "Resources": [
    "arn:*:fsx:*"
   ],
   "SelectionName": "FSxBackup"
  }
```

It basically seems like we're missing a "conditions" property on the [BackupSelection class](https://github.com/aws/aws-cdk/blob/v2.93.0/packages/aws-cdk-lib/aws-backup/lib/selection.ts).
